tests-ui/tests/store/modelToNodeStore.test.ts

```diff
@@ -0,0 +1,245 @@
+      // Optional chaining for consistency with getNodeProvider() return type
+      expect(retrieved?.key).toBe('custom_key')
+    })
+
+    it('should handle multiple providers for same model type and return first as primary', () => {
+      const provider1 = new ModelNodeProvider(
+        nodeDefStore.nodeDefsByName['UNETLoader'],
+        'key1'
+      )
+      const provider2 = new ModelNodeProvider(
+        nodeDefStore.nodeDefsByName['VAELoader'],
+        'key2'
+      )
+
+      store.registerNodeProvider('multi_type', provider1)
+      store.registerNodeProvider('multi_type', provider2)
+
+      const allProviders = store.getAllNodeProviders('multi_type')
+      expect(allProviders).toHaveLength(2)
+      expect(store.getNodeProvider('multi_type')).toStrictEqual(provider1)
+    })
+
+    it('should initialize new model type when first provider is registered', () => {
+      expect(store.modelToNodeMap['new_type']).toBeUndefined()
+
+      const provider = new ModelNodeProvider(
+        nodeDefStore.nodeDefsByName['UNETLoader'],
+        'test_key'
+      )
+      store.registerNodeProvider('new_type', provider)
+
+      expect(store.modelToNodeMap['new_type']).toBeDefined()
+      expect(store.modelToNodeMap['new_type']).toHaveLength(1)
+    })
+  })
+
+  describe('quickRegister', () => {
+    it('should connect node class to model type with parameter mapping', () => {
+      store.quickRegister('test_type', 'UNETLoader', 'test_param')
+
+      const provider = store.getNodeProvider('test_type')
+      expect(provider).toBeDefined()
+      // Using optional chaining since getNodeProvider() can return undefined
+      expect(provider?.nodeDef.name).toBe('UNETLoader')
+      expect(provider?.key).toBe('test_param')
+    })
+
+    it('should handle registration of non-existent node classes gracefully', () => {
+      expect(() => {
+        store.quickRegister('test_type', 'NonExistentLoader', 'test_param')
+      }).not.toThrow()
+
+      const provider = store.getNodeProvider('test_type')
+      // Optional chaining needed since getNodeProvider() can return undefined
+      expect(provider?.nodeDef).toBeUndefined()
+    })
+
+    it('should allow multiple node classes for same model type', () => {
+      store.quickRegister('multi_type', 'UNETLoader', 'param1')
+      store.quickRegister('multi_type', 'VAELoader', 'param2')
+
+      const providers = store.getAllNodeProviders('multi_type')
+      expect(providers).toHaveLength(2)
+    })
+  })
+
+  describe('registerDefaults integration', () => {
+    it('should register all expected model types based on mock data', () => {
+      store.registerDefaults()
+
+      const expectedTypes = [
+        'checkpoints',
+        'loras',
+        'vae',
+        'controlnet',
+        'unet',
+        'upscale_models',
+        'style_models',
+        'gligen'
+      ]
+
+      expectedTypes.forEach((modelType) => {
+        expect(store.getNodeProvider(modelType)).toBeDefined()
```

Comment:
When you're looping like this, you usually want to [add .soft](https://vitest.dev/api/expect.html#soft) so that you don't have to keep re-running it if there are multiple violations.